### PR TITLE
[cryptopp] Update to version 8.6.0 to fix CVE-2021-40530

### PR DESCRIPTION
**Pull request by isanych**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes https://nvd.nist.gov/vuln/detail/CVE-2021-40530

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Triplets not changed, tested on windows and linux; Baseline not updated

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/cryptopp/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cryptopp",
-  "version-string": "8.5.0",
+  "version-string": "8.6.0",
   "port-version": 1,
```
Comment (JackBoosY) [inline ports/cryptopp/vcpkg.json:4]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please run command `vcpkg x-add-version --all` then commit the changes.

File: ports/cryptopp/cmake.patch [hunk 2]
```diff
@@ -13,3 +13,17 @@ index 1683961..f63c71e 100644
  # Enable PIC for all target machines except 32-bit i386 due to register pressures.
  if (NOT CRYPTOPP_I386)
    SET(CMAKE_POSITION_INDEPENDENT_CODE 1)
+@@ -1022,10 +1026,10 @@ endif ()
+ # add_compile_definitions added in CMake 3.12
+ if (${CMAKE_VERSION} VERSION_LESS "3.12")
+   # https://stackoverflow.com/q/61250087
+-  add_definitions("${CMAKE_CPP_FLAGS}" "${CMAKE_CXX_FLAGS}" "${CRYPTOPP_COMPILE_DEFINITIONS}" "${CRYPTOPP_COMPILE_OPTIONS}")
++  add_definitions(${CMAKE_CPP_FLAGS} ${CRYPTOPP_COMPILE_DEFINITIONS} ${CRYPTOPP_COMPILE_OPTIONS})
```
Comment (JackBoosY) [inline ports/cryptopp/cmake.patch:21]: I think we should keep the double quotes, is it?

File: ports/cryptopp/cmake.patch [hunk 3]
```diff
@@ -13,3 +13,17 @@ index 1683961..f63c71e 100644
...
+ else()
+-  add_compile_definitions("${CMAKE_CPP_FLAGS}" "${CRYPTOPP_COMPILE_DEFINITIONS}")
+-  add_compile_options("${CMAKE_CXX_FLAGS}" "${CRYPTOPP_COMPILE_OPTIONS}")
++  add_compile_definitions(${CMAKE_CPP_FLAGS} ${CRYPTOPP_COMPILE_DEFINITIONS})
++  add_compile_options(${CRYPTOPP_COMPILE_OPTIONS})
```
Comment (JackBoosY) [inline ports/cryptopp/cmake.patch:26]: Same as above.

File: ports/cryptopp/vcpkg.json [hunk 4]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "cryptopp",
-  "version-string": "8.5.0",
-  "port-version": 1,
+  "version-string": "8.6.0",
```
Comment (JackBoosY) [inline ports/cryptopp/vcpkg.json:3]:
```suggestion
  "version": "8.6.0",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/cryptopp/cmake.patch (hunk 2 above)
Comment (isanych) [reply ports/cryptopp/cmake.patch:21]: quotes are incorrect there - they are messing up multiple options, I've created PR upstream too - https://github.com/noloader/cryptopp-cmake/pull/79

File: ports/cryptopp/vcpkg.json (hunk 4 above)
Comment (JackBoosY) [reply ports/cryptopp/vcpkg.json:3]: Please consider modify this.

File: versions/c-/cryptopp.json [hunk 5]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "3c5414e3593952556df042caa2b23b93f72f4698",
```
Comment (JackBoosY) [inline versions/c-/cryptopp.json:4]:
```suggestion
      "git-tree": "0d47535ac1767600626347e025079450a909c6ba",
```

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**